File: Homework/Mary_Antonyan/C++/20_03_2017/soldier.cpp
#include "soldier.hpp"

//Constructors
Soldier::Soldier() {}

Soldier::Soldier(const std::string& name, const std::string& weapon) 
    : m_name(name)
    , m_weapon(weapon) 
{ }

//Destructor
Soldier::~Soldier() { }

//Set methods
void Soldier::setName(const std::string& name) {
    m_name = name;
}

void Soldier::setWeapon(const std::string& weapon) {
    m_weapon = weapon;
}

//Get methods
std::string Soldier::getName() const {
    return m_name;
}

std::string Soldier::getWeapon() const {
    return m_weapon;
}

//Method
void Soldier::fight(const Soldier& enemy) {
    std::cout << "\n----------FIGHT----------\n";
    for (int i = 0; 5 > i; ++i) {
        std::cout << "....." << std::endl;
        
    }
    std::cout << m_name << " killed " << enemy.getName() << std::endl; 
}

File: Homework/Mary_Antonyan/Java/02_08_2017/Classes/Soldier.java
package Classes;

public class Soldier {
    // Members
    private String name;
    private String nationality;
    private int radius;
    private int health;
    private int x;
    private int y;
    private Weapon weapon;

    // Constructor
    public Soldier(String name, String nationality, int radius, int health, int x, int y, Weapon weapon) {
        setName(name);
        setNationality(nationality);
        setRadius(radius);
        setHealth(health);
        setX(x);
        setY(y);
        setWeapon(weapon);
    }

    // Methods
    public void setName(String name) {
        // TODO: add validation for name
        this.name = name;
    }

    public void setNationality(String nationality) {
        // TODO: add validation for nationality
        this.nationality = nationality;
    }

    public void setRadius(int radius) {
        this.radius = (radius > 1) ? radius : 1;
    }

    public void setHealth(int health) {
        this.health = (health > 0) ? health : 100;
    }

    public void setX(int x) {
        this.x = (x > 0) ? x : 0;
    }

    public void setX(int y) {
        this.y = (y > 0) ? y : 0;
    }

    public void setWeapon(Weapon weapon) {
        this.weapon = new Weapon(weapon);
    }

    public String getName() {
        return name;
    }

    public String getNationality() {
        return nationality;
    }

    public int getRadius() {
        return radius;
    }

    public int getHealth() {
        return health;
    }

    public Weapon getWeapon() {
        return weapon;
    }

    public void shoot() {
        // TODO: Add statements to this function to implement shoot action
         
    }
}
